pyked/converters.py

```diff
@@ -0,0 +1,522 @@
+             )
+        reference['detail'] = ref_key
+        if reference['detail'][-1] != '.':
+            reference['detail'] += '.'
+    else:
+        # Need one of DOI or preferredKey
+        raise MissingAttributeError('preferredKey', 'bibliographyLink')
+
+    return reference
+
+
+def get_experiment_kind(root):
+    """Read common properties from root of ReSpecTh XML file.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with experiment type and apparatus information.
+    """
+    properties = {}
+    if root.find('experimentType').text == 'Ignition delay measurement':
+        properties['experiment-type'] = 'ignition delay'
+    else:
+        #TODO: support additional experimentTypes
+        raise NotImplementedError(root.find('experimentType').text + ' not (yet) supported')
+
+    properties['apparatus'] = {'kind': '', 'institution': '', 'facility': ''}
+    try:
+        kind = root.find('apparatus/kind').text
+    except:
+        raise MissingElementError('apparatus/kind')
+    if kind in ['shock tube', 'rapid compression machine']:
+        properties['apparatus']['kind'] = kind
+    else:
+        raise NotImplementedError(kind + ' experiment not (yet) supported')
+
+    return properties
+
+
+def get_common_properties(root):
+    """Read common properties from root of ReSpecTh XML file.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with common properties
+    """
+    properties = {}
+
+    for elem in root.iterfind('commonProperties/property'):
+        name = elem.attrib['name']
+
+        if name == 'initial composition':
+            properties['composition'] = {'species': []}
+            composition_type = None
+
+            for child in elem.iter('component'):
+                spec = {}
+                spec['species-name'] = child.find('speciesLink').attrib['preferredKey']
+
+                # use InChI for unique species identifier (if present)
+                try:
+                    spec['InChI'] = child.find('speciesLink').attrib['InChI']
+                except KeyError:
+                    # TODO: add InChI validator/search
+                    warn('Missing InChI for species ' + spec['species-name'])
+                    pass
+
+                # amount of that species
+                spec['amount'] = [float(child.find('amount').text)]
+
+                properties['composition']['species'].append(spec)
+
+                # check consistency of composition type
+                if not composition_type:
+                    composition_type = child.find('amount').attrib['units']
+                elif composition_type != child.find('amount').attrib['units']:
+                    raise KeywordError('inconsistent initial composition units')
+            assert composition_type in ['mole fraction', 'mass fraction']
```

Comment:
We can at least support conversion of all the three that ChemKED supports, and modify if we encounter ReSpecTh files defined differently